### PR TITLE
Bind charm compiler's location at configuration time

### PR DESCRIPTION
**Pull request by NK-Nikunj**
Currently, the CMake build uses CHARM_HOME environment variable but does not look for the same at configuration time. This can lead to the following issues:
 1. If the variable is not set, the build system uses /bin/charmc as the default location which for obvious reasons can't be the correct location.
 2. If the variable is changed between generation and build process, this can lead to silent errors. While this can't occur with the current build system, an extension may later lead to this error.

This commit fixes it by doing the following:
 1. CHARM_HOME is detected and embedded at configuration time. This means that any user-defined variables (not exported) will have to be explicitly provided at the CMake step. This mitigates errors where a variable is only defined in the user scope and therefore not detected during the build process (when doing make).
 2. Throw error at configuration step if charm compiler is not found at configuration time. This leads to better error detection and propagation.

### CODE REVIEW

File: Makefile.common [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Points to CHARM_HOME
+CHARMC=/home/nk/src/charm/bin/charmc
+HYPERCOMM_HOME=/home/nk/src/hypercomm
```
Comment (jszaday) [inline Makefile.common:3]: These are user-specific paths -- how can they be generalized?

Comment (NK-Nikunj) [reply Makefile.common:3]: Wait this file should not exist in the PR. This file is generated from `Makefile.common.in` using cmake's `configure_file` facility. Let me correct it.

As for generalizing it: see https://github.com/jszaday/hypercomm/blob/cmake/CMakeLists.txt#L26. This takes the template file and generates user-specific paths.

**Review by jszaday — APPROVED** [review]
Great, LGTM. Thanks for being the first contributor :)

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,8 +2,16 @@ cmake_minimum_required(VERSION 3.10)
 
 project(hypercomm VERSION "0.0.001")
 
-set(CMAKE_C_COMPILER "$(CHARM_HOME)/bin/charmc")
-set(CMAKE_CXX_COMPILER "$(CHARM_HOME)/bin/charmc")
+
+set(CHARMC_PATH $ENV{CHARM_HOME})
+if(CHARMC_PATH)
+    message(STATUS "Charm compiler found at: ${CHARMC_PATH}")
```
Comment (NK-Nikunj) [inline CMakeLists.txt:8]: Let me know if you want better messages. I wrote them on a whim.

File: CMakeLists.txt [hunk 3]
```diff
@@ -2,8 +2,16 @@ cmake_minimum_required(VERSION 3.10)
...
+else(CHARMC_PATH)
+    message(FATAL_ERROR "Charm compiler not found. Set path using environment variable CHARM_HOME to the right location.")
```
Comment (jszaday) [inline CMakeLists.txt:10]:
```suggestion
    message(FATAL_ERROR "Charm compiler not found, please update the environment variable CHARM_HOME to the right location.")
```

Comment (jszaday) [reply CMakeLists.txt:10]: haha, well if I were to suggest anything it would be something like this

**Review by jszaday — APPROVED** [review]

**Pull request merged**